kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/ProduceAction.java

```diff
@@ -118,9 +118,17 @@ public void produce(
         .from(requests)
         .compose(
             request -> {
-              Optional<Duration> waitFor =
-                  produceRateLimiters.calculateGracePeriodExceeded(clusterId);
-              return produce(clusterId, topicName, request, controller, waitFor);
+              Optional<Long> messageSize = request.getOriginalSize();
+              if (messageSize.isPresent()) {
```

Comment:
See https://github.com/confluentinc/kafka-rest/pull/941, on how to remove this check.

Reply:
Thanks for pulling that other PR together.  I'm thinking we should keep the change under this PR as small as possible for now, for the EA cherry pick.  We've done so much testing on the code we've got already it seems risky to me to change it any more than is absolutely necessary at this point.

Reply:
Ack.